src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -6782,5 +6782,80 @@ static void Main()
             Assert.Equal(RefKind.In, lambdas[1].Parameters[0].RefKind);
             Assert.Equal(RefKind.Out, lambdas[2].Parameters[0].RefKind);
         }
+
+        [Fact]
+        public void StaticPartialLambda()
```

Comment:
If in future the parser allowed `partial`, we should update this code path:

https://github.com/dotnet/roslyn/blob/7693a6fec1fb518981df205bf784aaf9d4b8995c/src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs#L307 to report ERR_PartialMisplaced.

This test makes sure this will not be forgotten.